### ISSUE TITLE
Error: 'No known conditions for "./react" specifier in "next-auth" package' when testing with Jest

### ISSUE THREAD

**Issue by mariazab** [nextauthjs/next-auth#8328]
### Environment

OS: Linux 6.2 Ubuntu 23.04 23.04 (Lunar Lobster)
Node: 18.17.1

"next": "13.4.16",
"next-auth": "4.23.0",
"react": "18.2.0",
"jest": "29.6.2",

### Reproduction URL

https://github.com/mariazab/next-auth-jest-error

### Describe the issue

After updating NextAuth to 4.23.0 our tests fail with 

```
Test suite failed to run

    No known conditions for "./react" specifier in "next-auth" package

      1 | import "@testing-library/jest-dom";
      2 | import { render, screen } from "@testing-library/react";
    > 3 | import { useSession } from "next-auth/react";
        | ^
      4 | import Index from "../pages";
      5 |
      6 | it("renders status", async () => {

      at e (.yarn/cache/resolve.exports-npm-2.0.2-f59b42bbe5-1c7778ca1b.zip/node_modules/resolve.exports/dist/index.js:1:25)
      at Object.<anonymous> (test/index.test.tsx:3:1)

 ```
 
 This is only an issue with the tests, otherwise the application works.

### How to reproduce

1. Clone the repository
2. Run `yarn install`
3. Run `yarn test`

### Expected behavior

No error in testing.

**Comment by gvillenave** [comment]
We are seeing the same thing when updating from 4.22.4 to 4.23.0. I suspect this is due to commit 05ff6ae221b0434784e056de4bff6e3e45c49552 that changed the export structure, and somehow that must cause `babel-jest` to ignore those files.

**Comment by idimetrix** [comment]
Module not found: Package path ./react is not exported from package the same problem!

**Comment by ThangHuuVu** [comment]
I just released 4.23.1 that includes the fix for this issue, please check it out. Thanks for reporting & fixing it!

**Comment by mariazab** [comment]
That works, thanks!
